### DISCUSSION TITLE
0 Devices Connected After Migrating

### DISCUSSION THREAD

**Discussion by Pashovski** [Koenkk/zigbee2mqtt#6928] [Devices]
I'm in the process of migrating from my Pi running Home Assistant OS > NUC running Home Assistant container using docker-compose. My coordinator is a ZZH.  My log shows 0 devices connected

Here's is what I have done in the migration process so far:

- copied most of the settings; pan_id, network_key, channel, etc from my HA Addon to my new configuration.yaml in my z2m directory on my new host.
- copied these files: devices.yaml , configuration.yaml , state.json , from my HA addon to my new host
- made sure to ad ``listener 1883`` to ``mosquitto.conf``

log: https://hatebin.com/trqaqrsmps
working settings: https://paste.ubuntu.com/p/DWp5GsZYpT/
NUC settings: https://paste.ubuntu.com/p/4TjShs8GDR/
mosquitto.conf: https://paste.ubuntu.com/p/7jkMKSTq7y/

**Comment by code-in-progress** [comment]
You need to also copy over the files in your data/ directory. Specifically `database.db`.

**Reply by Pashovski** [reply]
I copied database.db . Forgot to mention that, sorry.

Eveything that was in my share/zigbee2mqtt directory from my Pi , is now in my zigbee2mqtt/data directory in my nuc

**Reply by code-in-progress** [reply]
Looks like it's a problem with 1.18.2. I just upgraded and while all my devices are connecting to MQTT, they aren't showing up in the z2m frontend: https://github.com/Koenkk/zigbee2mqtt/issues/6918

**Reply by code-in-progress** [reply]
So, I just fixed my frontend device problem by running `npm ci --production` after doing a `git pull`. That might fix your problem as well.

**Reply by Pashovski** [reply]
Great news! Can you tell me exactly how to do what you just did? I'm fairly new to Linux

**Reply by code-in-progress** [reply]
Sure. Go into your z2m directory (I have mine at `/srv/zigbee2mqtt`, but by default I think it is at `/opt/zigbee2mqtt`). Backup your data directory (which you seem to already have done). Then, issue a `git pull`. That will pull in all the latest fixes. After that, run `npm ci --production`. It might take a few minutes to run. After it completes, test your install using `npm start`.

That's for a bare metal install. If you are running the docker version, I'm not going to be able to help very much.

**Reply by Tediore** [reply]
Docker would likely just involve pulling the `latest-dev` image which hopefully contains the fixes.

**Comment by Pashovski** [comment]
It was a transfer error. When I copied the database.db from the Pi to the NUC over SCP, only the coordinator was in the database, and the 24 devices were left out. I manually copied the entries and it's working.
